db/builder.cc

```diff
@@ -179,7 +179,7 @@ Status BuildTable(
         &snapshots, earliest_write_conflict_snapshot, snapshot_checker, env,
         ShouldReportDetailedTime(env, ioptions.statistics),
         true /* internal key corruption is not ok */, range_del_agg.get(),
-        blob_file_builder.get());
+        blob_file_builder.get(), ioptions.log_corruption_errors);
```

Comment:
I wonder if we should use a more generic name, such as `redact_data_from_errors` or (inversely) `allow_data_in_errors`. Then, in addition to determining whether data can appear in logged errors, it can also determine whether data appears in `Status`es returned via our user API.